File: frontend/src/app/models/grupo.ts
import { IEstatus } from "./estatus";
import { IMateria } from "./materia";

export interface IGrupo extends IEstatus{
    AlumnosInscritos: number;
    Materia: IMateria | null,
    CantidadAlumnos: number,
    Nombre: string,
    IdGrupo: number,
    Clave: number
}

export class Grupo implements IGrupo{
    Materia: IMateria | null = null;
    CantidadAlumnos: number = 0;
    IdGrupo: number = 0;
    Clave: number = 0;
    AlumnosInscritos: number = 0;
    Nombre: string = "";
    Estatus: string = "";
}

File: backend/src/main/java/models/Grupo.java
package models;

public class Grupo extends Materia{
	
	private int IdGrupo;
	private int CantidadAlumnos = 0;
	
	private int AlumnosInscritos = 0;

	public int getCantidadAlumnos() {
		return CantidadAlumnos;
	}

	public void setCantidadAlumnos(int cantidadAlumnos) {
		this.CantidadAlumnos = cantidadAlumnos;
	}

	public int getIdGrupo() {
		return IdGrupo;
	}

	public void setIdGrupo(int idGrupo) {
		IdGrupo = idGrupo;
	}

	public int getAlumnosInscritos() {
		return AlumnosInscritos;
	}

	public void setAlumnosInscritos(int alumnosInscritos) {
		AlumnosInscritos = alumnosInscritos;
	}
	
	
}
